sycl/include/CL/sycl/builtins.hpp

```diff
@@ -1548,99 +1441,86 @@ namespace half_precision {
 }
 
 // genfloatf exp2 (genfloatf x)
 template <typename T>
-typename std::enable_if<detail::is_genfloatf<T>::value, T>::type
-exp2(T x) __NOEXC {
+detail::enable_if_t<detail::is_genfloatf<T>::value, T> exp2(T x) __NOEXC {
   return __sycl_std::__invoke_half_exp2<T>(x);
 }
 
 // genfloatf exp10 (genfloatf x)
 template <typename T>
-typename std::enable_if<detail::is_genfloatf<T>::value, T>::type
-exp10(T x) __NOEXC {
+detail::enable_if_t<detail::is_genfloatf<T>::value, T> exp10(T x) __NOEXC {
   return __sycl_std::__invoke_half_exp10<T>(x);
 }
 
 // genfloatf log (genfloatf x)
 template <typename T>
-typename std::enable_if<detail::is_genfloatf<T>::value, T>::type
-log(T x) __NOEXC {
+detail::enable_if_t<detail::is_genfloatf<T>::value, T> log(T x) __NOEXC {
   return __sycl_std::__invoke_half_log<T>(x);
 }
 
 // genfloatf log2 (genfloatf x)
 template <typename T>
-typename std::enable_if<detail::is_genfloatf<T>::value, T>::type
-log2(T x) __NOEXC {
+detail::enable_if_t<detail::is_genfloatf<T>::value, T> log2(T x) __NOEXC {
   return __sycl_std::__invoke_half_log2<T>(x);
 }
 
 // genfloatf log10 (genfloatf x)
 template <typename T>
-typename std::enable_if<detail::is_genfloatf<T>::value, T>::type
-log10(T x) __NOEXC {
+detail::enable_if_t<detail::is_genfloatf<T>::value, T> log10(T x) __NOEXC {
   return __sycl_std::__invoke_half_log10<T>(x);
 }
 
 // genfloatf powr (genfloatf x, genfloatf y)
 template <typename T>
-typename std::enable_if<detail::is_genfloatf<T>::value, T>::type
-powr(T x, T y) __NOEXC {
+detail::enable_if_t<detail::is_genfloatf<T>::value, T> powr(T x, T y) __NOEXC {
   return __sycl_std::__invoke_half_powr<T>(x, y);
 }
 
 // genfloatf recip (genfloatf x)
 template <typename T>
-typename std::enable_if<detail::is_genfloatf<T>::value, T>::type
-recip(T x) __NOEXC {
+detail::enable_if_t<detail::is_genfloatf<T>::value, T> recip(T x) __NOEXC {
   return __sycl_std::__invoke_half_recip<T>(x);
 }
 
 // genfloatf rsqrt (genfloatf x)
 template <typename T>
-typename std::enable_if<detail::is_genfloatf<T>::value, T>::type
-rsqrt(T x) __NOEXC {
+detail::enable_if_t<detail::is_genfloatf<T>::value, T> rsqrt(T x) __NOEXC {
   return __sycl_std::__invoke_half_rsqrt<T>(x);
 }
 
 // genfloatf sin (genfloatf x)
 template <typename T>
-typename std::enable_if<detail::is_genfloatf<T>::value, T>::type
-sin(T x) __NOEXC {
+detail::enable_if_t<detail::is_genfloatf<T>::value, T> sin(T x) __NOEXC {
   return __sycl_std::__invoke_half_sin<T>(x);
 }
 
 // genfloatf sqrt (genfloatf x)
 template <typename T>
-typename std::enable_if<detail::is_genfloatf<T>::value, T>::type
-sqrt(T x) __NOEXC {
+detail::enable_if_t<detail::is_genfloatf<T>::value, T> sqrt(T x) __NOEXC {
   return __sycl_std::__invoke_half_sqrt<T>(x);
 }
 
 // genfloatf tan (genfloatf x)
 template <typename T>
-typename std::enable_if<detail::is_genfloatf<T>::value, T>::type
-tan(T x) __NOEXC {
+detail::enable_if_t<detail::is_genfloatf<T>::value, T> tan(T x) __NOEXC {
```

Comment:
Arrived here I have an overdose of `detail::enable_if_t<detail::is_genfloatf<T>::value, T>` :-)
What about defining a meta-function with `using` to factorize out this?
Add as many meta-function you can to reduce the code size...

Reply:
Hi, I agree with your comment, but I'd prefer not to inflate this patch any further.
In the near future I will create a couple of patches for this functionality and I will apply your suggestion in them.
Could you provide a small code snippet of what you would like this code to look like?

Reply:
We need to find a name understandable by native English speakers, which I am not...
A few ideas
```
detail::enable_if_genfloat_t<T> tan(T x) __NOEXC {
detail::only_genfloat_t<T> tan(T x) __NOEXC {
detail::only<detail::is_genfloatf, T> tan(T x) __NOEXC {
```